EventSubscriber/FetchMetadataRequestSubscriber.php

```diff
@@ -32,14 +37,33 @@ public function requestEvent(RequestEvent $event): void
         $request = $event->getRequest();
         $options = $this->configProvider->getPathConfig($request);
         $fetchMetadataPolicy = $this->fetchMetadataPolicyProvider->getFetchMetadataPolicy($options['fetch_metadata']);
-
         if ($options['fetch_metadata']['active']) {
             $response = new Response();
             $response->headers->set('Vary', 'sec-fetch-site, sec-fetch-dest, sec-fetch-mode');
+
             if (!$fetchMetadataPolicy->applyPolicy($request)) {
+                $this->logRejection($request, $options['fetch_metadata']['policy']);
+
                 $response->setStatusCode(Response::HTTP_UNAUTHORIZED);
                 $event->setResponse($response);
             }
         }
     }
+    /**
+     * Log rejection function. Logs if a request has been rejected by the fetch metadata policy.
+     * Records the Url, and which policy rejected the request.
+     *
+     * @param String $policy, A reference string to the Fetch Metadata policy used in the request. If Null assumed Default Policy
+     * @param Request $request, The request
+     * @return void
+     */
+    private function logRejection(Request $request, $policy): void
+    {
+        $debugContext = [
+            'request' => $request->getContent(),
+            'policy' => $policy
+        ];
+        $fetchMetaPolicy = $policy ?? 'Default Policy';
+        $this->logger->debug($request->getMethod().' request from host '.$request->getUri().' blocked by Fetch Metadata Policy: '.$fetchMetaPolicy, $debugContext);
```

Comment:
Put the log message "template" into a separate constant or member variable and format it for [`sprintf()`](https://www.php.net/sprintf). e.g. something like:

```php
private $requestBlockedMsg = '%1$s request from host %2$s blocked by Fetch Metadata Policy: %3$s';
// [...]
$msg =sprintf($this->requestBlockedMsg, $request->getMethod(), $request->getUri(), $fetchMetaPolicy);
```

Makes it easier to handle escaping, formatting, translation (later).

Reply:
Oh okay cool. Yeah I'll try and apply that same format in other places that we're handling logging too. Thanks